src/Cake.Common/Tools/VSWhere/VSWhereAliases.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cake.Common.Tools.VSWhere.All;
+using Cake.Common.Tools.VSWhere.Latest;
+using Cake.Common.Tools.VSWhere.Legacy;
+using Cake.Common.Tools.VSWhere.Product;
+using Cake.Core;
+using Cake.Core.Annotations;
+
+namespace Cake.Common.Tools.VSWhere
+{
+    /// <summary>
+    /// Contains functionality related to executing VSWhere tool.
+    /// </summary>
+    [CakeAliasCategory("VSWhere")]
+    public static class VSWhereAliases
+    {
+        /// <summary>
+        /// Gets the legacy Visual Studio product installation paths.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="latest">Get the latest version.</param>
+        /// <returns>The Visual Studio installation path.</returns>
+        /// <example>
+        /// <code>
+        ///     var legacyInstallationPath = VSLegacy(true);
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Legacy")]
+        [CakeNamespaceImport("Cake.Common.Tools.VSWhere.Legacy")]
+        public static string VSLegacy(this ICakeContext context, bool latest)
```

Comment:
The purpose according to the documentation is to return the newest version or last installed and includes 2015 or older instead of just being limited to VS2017.  Latest only searches 2017 from my understanding.